theano/sandbox/cuda/opt.py

```diff
@@ -1867,6 +1867,27 @@ def local_gpu_downsample_factor_max_grad(node):
                                               as_cuda_ndarray_variable(z),
                                               as_cuda_ndarray_variable(gz)))]
 
+@register_opt()
+@local_optimizer([downsample.DownsampleFactorMaxRop])
+def local_gpu_downsample_factor_max_rop(node):
+    if (isinstance(node.op, downsample.DownsampleFactorMaxRop) and
+        node.op.ds == node.op.st):
+
+        assert node.op.__props__ == ('ds', 'ignore_border', 'st', 'padding',
+                                     'mode')
+        if (node.op.padding != (0, 0) or
+            node.op.mode != 'max' or
+            node.op.st != node.op.ds):
```

Comment:
that seem the good update for now compared to the original code in the other PR.